### PR TITLE
Certificate configuration 'gotchas'

### PR DESCRIPTION
**Pull request by Kelketek**
Adds in notes for a couple of gotchas found while trying to configure certificates.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Kelketek! I've created [OSPR-5343](https://openedx.atlassian.net/browse/OSPR-5343) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Kelketek** [comment]
It should be noted that there is a PR to fix some of this at https://github.com/edx/edx-certificates/pull/49 and that even if said PR was merged, Cypress users would not reap the benefit.

@bradenmacdonald could you give this a look over?

@antoviaque Does this and the other PR count as OSPR?

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 1]
```diff
@@ -43,6 +43,10 @@ Enable Certificates in Studio and the Learning Management System
 
 #. Save the ``/cms/envs/common.py`` and ``/lms/envs/common.py`` files.
 
+#. If it does not exist already, create the folder `/tmp/certificates` owned by
```
Comment (mhoeber) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: double `-``/tmp/certificates``

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 2]
```diff
@@ -43,6 +43,10 @@ Enable Certificates in Studio and the Learning Management System
...
+   the user and group ``www-data``. Depending on your configuration, this folder 
+   may not survive reboots, and so may need to be created by script.
```
Comment (mhoeber) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: replace "may" with "might" 2x

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 3]
```diff
@@ -176,6 +180,12 @@ for eligible learners.
    For example, 
 
    ``/manage.py lms --settings=aws ungenerated_certs -c course-v1:edX+demoX_Demo_2015``.
+   
+   .. Note:: 
+     If you're on a server that does not have https support for the lms
```
Comment (mhoeber) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: "If the LMS is running on a server that does not have https support . . .

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 4]
```diff
@@ -176,6 +180,12 @@ for eligible learners.
    For example, 
 
    ``/manage.py lms --settings=aws ungenerated_certs -c course-v1:edX+demoX_Demo_2015``.
+   
+   .. Note:: 
+     If you're on a server that does not have https support for the lms 
+     (such as a locally run fullstack for testing) you will need to use the ``--insecure--``
+     flag so that the certificate generation service knows to contact the lms on http instead of
```
Comment (mhoeber) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: "knows to contact" -> "contacts"

**Comment by mhoeber** [comment]
@Kelketek thank you, a few minor nits

**Comment by Kelketek** [comment]
@mhoeber Notes addressed :)

**Comment by mhoeber** [comment]
:+1:

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 5]
```diff
@@ -43,6 +43,10 @@ Enable Certificates in Studio and the Learning Management System
 
 #. Save the ``/cms/envs/common.py`` and ``/lms/envs/common.py`` files.
 
+#. If it does not exist already, create the folder ``/tmp/certificates`` owned by
+   the user and group ``www-data``. Depending on your configuration, this folder 
+   might not survive reboots, and so might need to be created by script.
+
```
Comment (bradenmacdonald) [inline en_us/install_operations/source/configuration/enable_certificates.rst:49]: @Kelketek Any idea why this is not being done by [Ansible](https://github.com/edx/configuration/blob/master/playbooks/roles/certs/tasks/main.yml) or by the certificates app itself?

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 6]
```diff
@@ -176,6 +180,12 @@ for eligible learners.
    For example, 
 
    ``/manage.py lms --settings=aws ungenerated_certs -c course-v1:edX+demoX_Demo_2015``.
```
Comment (bradenmacdonald) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: These manage.py commands are all slightly wrong... They need to start with `./`, not `/`. That isn't your change but would be a nice fix while you're at it.

File: en_us/install_operations/source/configuration/enable_certificates.rst (hunk 5 above)
Comment (bradenmacdonald) [inline en_us/install_operations/source/configuration/enable_certificates.rst:49]: never mind - I realized that's what your other PR is for :)

File: en_us/install_operations/source/configuration/enable_certificates.rst [hunk 7]
```diff
@@ -176,6 +180,12 @@ for eligible learners.
...
+   
+   .. Note:: 
+     If the LMS is running on a server that does not have https support 
+     (such as a locally run fullstack for testing) you will need to use the ``--insecure--``
```
Comment (bradenmacdonald) [inline en_us/install_operations/source/configuration/enable_certificates.rst]: I think the parameter is just `--insecure`, isn't it? I tested and that does seem to be the case.

**Comment by Kelketek** [comment]
@bradenmacdonald Your notes should be addressed now. :)

**Comment by bradenmacdonald** [comment]
:+1: 

BTW, I assume we will be removing that note about creating the tmp dir once the next version of edx-platform is released and that fix is included, right?

**Comment by Kelketek** [comment]
@bradenmacdonald That should be done, yes. Though I'm not sure where to make note of it. @catong do you know where we can make note of this? The problem should be resolved by the next release if the PR for certificates gets merged in.

**Comment by catong** [comment]
Thanks @Kelketek, @mhoeber has taken over the certificates documentation from me, and is now more familiar with the latest state of the certificates feature. It also looks as if he has already been involved with this PR, so I will leave it in his hands.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Kelketek! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-documentation&number=508

**Comment by nedbat** [comment]
Hi, sorry for the new comment on this old pull request. We're testing out re-scanning contributions, and our bot added a confusing message that you can ignore. We'll fix that. Sorry about the noise.

**Pull request merged**